digital_image_processing/convert_to_grayscale.py

```diff
@@ -0,0 +1,32 @@
+"""
+    Implemented an algorithm using opencv to convert a colored image into its grayscale
+"""
+from cv2 import destroyAllWindows, imread, imshow, waitKey
+
+
+def convert_to_grayscale(img):
+    # getting number of pixels in the image
```

Comment:
add below doctest-

    """
    >>> import cv2
    >>> import numpy as np
    >>> img = np.zeros((100, 100, 3), dtype=np.uint8)
    >>> img[:, :, 0] = 255
    >>> gray = convert_to_grayscale(img)
    >>> assert np.allclose(gray[:, :, 0], 0.114 * 255, rtol=1e-03, atol=1e-03)
    >>> assert np.allclose(gray[:, :, 1], 0.587 * 255, rtol=1e-03, atol=1e-03)
    >>> assert np.allclose(gray[:, :, 2], 0.299 * 255, rtol=1e-03, atol=1e-03)
    """


then add below in main:-
if __name__ == "__main__":
    import doctest
    doctest.testmod()

Reply:
What are these?

>>> assert np.allclose(gray[:, :, 0], 0.114 * 255, rtol=1e-03, atol=1e-03)
>>> assert np.allclose(gray[:, :, 1], 0.587 * 255, rtol=1e-03, atol=1e-03)
>>> assert np.allclose(gray[:, :, 2], 0.299 * 255, rtol=1e-03, atol=1e-03)

Are you able to add these things by yourself?

Reply:
Where should exactly "doctest.testmod()" be placed?

Reply:
under main.

Reply:
In the beginning or in the end?

Reply:
Thank you for your detailed explanation. I added doctest.